packages/mobile/src/transactions/send.ts

```diff
@@ -116,11 +122,103 @@ export function* sendTransaction(
   account: string,
   tag: string,
   txId: string,
-  staticGas?: number | undefined
+  staticGas?: number,
+  cancelAction?: string
 ) {
-  const txPromises = yield call(sendTransactionPromises, tx, account, tag, txId, staticGas)
-  const confirmation = yield call(awaitConfirmation, txPromises)
-  return confirmation
+  const sendTxMethod = function*(nonce: number) {
+    const { confirmation } = yield call(
+      sendTransactionPromises,
+      tx,
+      account,
+      tag,
+      txId,
+      nonce,
+      staticGas
+    )
+    const result = yield confirmation
+    return result
+  }
+  yield call(wrapSendTransactionWithRetry, txId, account, sendTxMethod, cancelAction)
+}
+
+export function* wrapSendTransactionWithRetry(
```

Comment:
Working around the 3-promise based interface we're using in `sendTransactionPromises` turned out to be a pain. It has me rethinking if the promises are actually a nicer pattern in general then callbacks. They're great if consumers will be interacting with the results in a 'straight line' but not so nice for adding new middleware btwn the caller and sending utilities